### PR TITLE
[WM-1467] make Leo PR in update-and-publish action

### PR DESCRIPTION
**Pull request by mspector**
also checks for Jira ID before proceeding

### CODE REVIEW

File: .github/workflows/make_leo_pr.yml [hunk 1]
```diff
@@ -1,122 +0,0 @@
-name: Make Leonardo PR
-
-on:
-  # workflow_run:
-  #   workflows: [update-and-publish-cromwell-helm-chart]
-  #   types: [completed]
-  workflow_dispatch:
```
Comment (cjllanwarne) [inline .github/workflows/make_leo_pr.yml:7]: TOL I wonder whether leaving this action in place, but maybe with a manual trigger and no expectation of automerging on the other side, might be a good idea? It may be nice to be able to force creation of a update PR? OTOH it might be become yet another unused thing to maintain? And we can already recreate it if we need it... so ok, I've changed my mind, let's delete this for now...

File: .github/workflows/update-and-publish-helm-chart.yml [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
+    branches:
+      - 'main'
 
 jobs:
   update-and-publish:
     if: ( !startsWith(github.event.head_commit.message, 'Auto update cromwell-helm chart version') )
     name: Update and publish new cromwell-helm chart
     runs-on: ubuntu-latest
     steps:
+      - name: Ensure that a Jira ID is present in the commit message
```
Comment (cjllanwarne) [inline .github/workflows/update-and-publish-helm-chart.yml:13]: TOL:

- Can we have a github action that checks that PR titles have jira tags before allowing us to commit?
- Can we validate/change our repo settings to make sure the PR title becomes (at least part of) the commit message?

File: .github/workflows/update-and-publish-helm-chart.yml [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
+        run: |
+          JIRA_ID=$(echo '${{ github.event.head_commit.message }}' | grep -Eo '\[?[A-Z][A-Z]+-[0-9]+\]?')
```
Comment (cjllanwarne) [inline .github/workflows/update-and-publish-helm-chart.yml:15]: Some thoughts:
- Does this work on multi-line messages?
- Does this work if the jira ID is embedded in `[]`s? Or in other ways not at the start of the message?
  - If we eventually add a "Does the title contain a jira ID" check, we should duplicate the same pattern matching we use here to pick it out of the title.
- Do we want an allow-list of JIRA project codes we might use (to reduce the chance of falsely identifying a random part of the message string as a jira ID)?
- This is a great candidate for "if it fails, alert the team in slack"

File: .github/workflows/update-and-publish-helm-chart.yml [hunk 4]
```diff
@@ -57,3 +62,109 @@ jobs:
           git config --global user.email "broadbot@broadinstitute.org"
           git commit -am "Auto update cromwell-helm chart version to $NEXT_VERSION_GCP and coa-helm chart version to $NEXT_VERSION_COA"
           git push https://broadbot:$BROADBOT_TOKEN@github.com/broadinstitute/cromwhelm.git main
+      
+      - name: Find out latest chart version from cromwhelm repository
```
Comment (cjllanwarne) [inline .github/workflows/update-and-publish-helm-chart.yml:66]: Is this step different from `- name: Find out next chart version`? Should it become more like a "check the expected new chart has appeared in the charts directory" kind of an action?

File: .github/workflows/update-and-publish-helm-chart.yml [hunk 5]
```diff
@@ -57,3 +62,109 @@ jobs:
...
+          echo ::set-output name=NEW_CROMWELL_HELM_V::${NEW_CROMWELL_HELM_V}
+      
+      - name: Create leonardo PR
+        id: create-leo-pr
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.BROADBOT_TOKEN }}
+          script: |
+            const pull_response = await github.rest.pulls.create({
+              title: '$JIRA_ID Update Cromwell version to ${{ steps.update-and-push.outputs.NEW_CROMWELL_HELM_V }}',
```
Comment (cjllanwarne) [inline .github/workflows/update-and-publish-helm-chart.yml:151]: Curly braces are "normally a good idea" to delimit the variable name, the colon is "what I tend to do in my own PRs". Take or leave both at your discretion:
```suggestion
              title: '${JIRA_ID}: Update Cromwell version to ${{ steps.update-and-push.outputs.NEW_CROMWELL_HELM_V }}',
```

**Review by cjllanwarne — APPROVED** [review]
I focused mainly on the action flow itself, rather than "did you code up the specific actions correctly", since that is presumably mostly copy/pasted from elsewhere.

Looks good to me, a couple of TOL level comments.

**Review by salonishah11 — APPROVED** [review]

File: .github/workflows/make_leo_pr.yml (hunk 1 above)
Comment (mspector) [reply .github/workflows/make_leo_pr.yml:7]: I went through the same sequence of thoughts, for what it's worth. I agree that it's easy enough to recreate it...

File: .github/workflows/update-and-publish-helm-chart.yml (hunk 3 above)
Comment (mspector) [reply .github/workflows/update-and-publish-helm-chart.yml:15]: > Does this work on multi-line messages?
> Does this work if the jira ID is embedded in []s? Or in other ways not at the start of the message?

I believe: yes, yes, and yes, based on some hand tests that I did

> Do we want an allow-list of JIRA project codes we might use (to reduce the chance of falsely identifying a random part of the message string as a jira ID)?

Maybe? I'm on the fence... let's talk about this tomorrow

> This is a great candidate for "if it fails, alert the team in slack"

Agreed. I'll add it to that ticket

File: .github/workflows/update-and-publish-helm-chart.yml (hunk 4 above)
Comment (mspector) [reply .github/workflows/update-and-publish-helm-chart.yml:66]: > Should it become more like a "check the expected new chart has appeared in the charts directory" kind of an action?

It's only different in that it reads directly from the charts, rather than using the computed version number

I originally added it as part of a copy/paste from the other workflow, and saw the similarity with the earlier step, but then decided to keep it because it seemed useful for that reason alone... but I agree that it'd be even better to perform a check against the computed number.
